src/__tests__/fsm.tests.ts

```diff
@@ -0,0 +1,91 @@
+import fsm from '../lib/fsm';
+
+describe('fsm-tests', () => {
+  it('should export fsm', () => {
+    let sut = new fsm('initial', []);
+
+    expect(sut).not.toBe(undefined);
+    expect(sut).not.toBe(null);
+  });
+});
+
+describe('fsm-tests', () => {
+  it('Initial state is <initial>', () => {
+    const initialState = 'initial';
+
+    let sut = new fsm(initialState, []);
+
+    expect(sut.CurrentState()).toBe(initialState);
+  });
+});
+
+describe('fsm-tests', () => {
+  it('Can handle a unknown message without transition', () => {
+    const initialState = 'initial';
+
+    let sut = new fsm(initialState, []);
+    sut.handle('test-message');
+
+    expect(sut.CurrentState()).toBe(initialState);
+  });
+});
+
+describe('fsm-tests', () => {
+  it('Can handle a known message to transition', () => {
+    const initialState = 'initial';
+    const message = 'test-message';
+    const toState = 'test-state';
+
+    let sut = new fsm(initialState, [fsm.Transition(initialState, message, toState)]);
+
+    sut.handle(message);
+    expect(sut.CurrentState()).toBe(toState);
+  });
+});
+
+describe('fsm-tests', () => {
+  it('Can run a complex automata', () => {
```

Comment:
this is testing different cases, some of them duplicates of the above cases. Split in test cases:

 Should navigate to next state (given initial then initial -> b
 Should navigate from a state where it previously transitioned (given initial->b, then b->c)
And so on